src/libos/src/net/syscalls.rs

```diff
@@ -531,6 +531,56 @@ pub fn do_recvmsg(fd: c_int, msg_mut_ptr: *mut msghdr_mut, flags_c: c_int) -> Re
     }
 }
 
+pub fn do_sendmmsg(
+    fd: c_int,
+    msgvec_ptr: *mut mmsghdr,
+    vlen: c_uint,
+    flags_c: c_int,
+) -> Result<isize> {
+    debug!(
+        "sendmmsg: fd: {}, msg: {:?}, flags: 0x{:x}",
+        fd, msgvec_ptr, flags_c
+    );
+
+    from_user::check_ptr(msgvec_ptr)?;
+
+    let mut msgvec = unsafe { std::slice::from_raw_parts_mut(msgvec_ptr, vlen as usize) };
+    let flags = SendFlags::from_bits_truncate(flags_c);
+    let file_ref = current!().file(fd as FileDesc)?;
+
+    if let Ok(socket) = file_ref.as_host_socket() {
+        let send_count = msgvec
+            .iter_mut()
+            .filter_map(|mut mmsg| {
+                if !mmsg.msg_hdr.check_member_ptrs().is_ok() {
+                    return None;
+                }
+
+                let msg = unsafe {
+                    if let Ok(msg) = MsgHdr::from_c({ &mmsg.msg_hdr }) {
+                        msg
+                    } else {
+                        return None;
+                    }
+                };
+
+                socket
+                    .sendmsg(&msg, flags)
+                    .map(|bytes_sent| {
+                        mmsg.msg_len = bytes_sent as u32;
```

Comment:
This should be adding, isn't it?
```rust
mmsg.msg_len += bytes_sent
```

Reply:
No. It only indicates the length of single message.